packages/core/src/storage/migration/updates/0.2-0.3/connection.ts

```diff
@@ -0,0 +1,61 @@
+import type { BaseAgent } from '../../../../agent/BaseAgent'
+import type { ConnectionRecord } from '../../../../modules/connections'
+
+import { ConnectionType, ConnectionRepository } from '../../../../modules/connections'
+import { MediationRepository } from '../../../../modules/routing'
+
+/**
+ * Migrate the {@link ConnectionRecord} to a 0.3 compatible format.
+ *
+ * @param agent
+ */
+export async function migrateConnectionRecordToV0_3<Agent extends BaseAgent>(agent: Agent) {
+  agent.config.logger.info('Migrating connection records to storage version 0.3')
+  const connectionRepository = agent.dependencyManager.resolve(ConnectionRepository)
+  const mediationRepository = agent.dependencyManager.resolve(MediationRepository)
+
+  agent.config.logger.debug('Fetching all connection records from storage')
+  const allConnections = await connectionRepository.getAll(agent.context)
```

Comment:
I don't think we have to resolve _all_ connection records, only the one that are for mediation records. Reason is that the number of connections _could_ be quite large. We should probably add a way to do pagination.

So we can resolve all mediaton records. Then for each of the mediation records, fetch the connection record, add the needed tags and then save the connection record

Reply:
Since we are adding the `connectionTypes` directly to the record, iterating through all records would be necessary